### DISCUSSION TITLE
UsdMayaPrimWriter::Write should return bool indicating success or failure?

### DISCUSSION THREAD

**Discussion by jjzanin-SMS** [Autodesk/maya-usd#3491] [Ideas]
Hello,

We have many custom prim writers that inherit from `UsdMayaPrimWriter` and when there is an error during the `Write` function, we would like the export to fail. I propose that the `UsdMayaPrimWriter::Write` function returns a boolean indicating success or failure. When false is returned, the calling code returns and propagates the failure. The `UsdMayaPrimWriter::PostExport` could benefit from this as well.

**Comment by wallworm** [comment]
Thank you for this suggestion. Sorry it took so long to respond. We'll discuss this internally.
